### PR TITLE
8294866: [lworld] Array classes access flags should include identity bits

### PR DESCRIPTION
**Pull request by RogerRiggs**
Update java.lang.reflect.AccessFlag to return IDENTITY for array classes.
The access flag bits for arrays come from the VM, hence the changes to HotSpot.

---------
### Progress
- [x] Change must not contain extraneous whitespace

### Issue
 * [JDK-8294866](https://bugs.openjdk.org/browse/JDK-8294866): [lworld] Array classes access flags should include identity bits (**Bug** - P3)

### Reviewers
 * [Chen Liang](https://openjdk.org/census#liach) (@liach - no project role) ⚠️ Review applies to [9e0368fc](https://git.openjdk.org/valhalla/pull/1141/files/9e0368fc40607bd47071ee3fe0b9299850c61c8e)
 * [Dan Heidinga](https://openjdk.org/census#heidinga) (@DanHeidinga - no project role) ⚠️ Review applies to [9e0368fc](https://git.openjdk.org/valhalla/pull/1141/files/9e0368fc40607bd47071ee3fe0b9299850c61c8e)
 * [Vicente Romero](https://openjdk.org/census#vromero) (@vicente-romero-oracle - Committer) ⚠️ Review applies to [9e0368fc](https://git.openjdk.org/valhalla/pull/1141/files/9e0368fc40607bd47071ee3fe0b9299850c61c8e)
 * [Frederic Parain](https://openjdk.org/census#fparain) (@fparain - Committer) ⚠️ Review applies to [9e0368fc](https://git.openjdk.org/valhalla/pull/1141/files/9e0368fc40607bd47071ee3fe0b9299850c61c8e)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/valhalla.git pull/1141/head:pull/1141` \
`$ git checkout pull/1141`

Update a local copy of the PR: \
`$ git checkout pull/1141` \
`$ git pull https://git.openjdk.org/valhalla.git pull/1141/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1141`

View PR using the GUI difftool: \
`$ git pr show -t 1141`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/valhalla/pull/1141.diff">https://git.openjdk.org/valhalla/pull/1141.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/valhalla/pull/1141#issuecomment-2180886133)

### CODE REVIEW

File: src/java.base/share/classes/java/lang/reflect/AccessFlag.java [hunk 1]
```diff
@@ -215,7 +215,7 @@ public Set<Location> apply(ClassFileFormatVersion cffv) {
             new Function<ClassFileFormatVersion, Set<Location>>() {
                 @Override
                 public Set<Location> apply(ClassFileFormatVersion cffv) {
-                    return (cffv.compareTo(ClassFileFormatVersion.RELEASE_22) >= 0
+                    return (cffv.compareTo(ClassFileFormatVersion.RELEASE_23) >= 0
```
Comment (liach) [inline src/java.base/share/classes/java/lang/reflect/AccessFlag.java:218]: Should we just use `cffv == ClassFileFormatVersion.latest()`?

**Review by liach — APPROVED** [review]
Thanks for the latest preview tagging.

**Review by DanHeidinga — APPROVED** [review]
Looks right to me.  @fparain any concerns?

**Review by vicente-romero-oracle — APPROVED** [review]
lgtm

**Comment by vicente-romero-oracle** [comment]
just wondering now that you are modifying AccessFlag, should we change SUPER:
```
SUPER(0x0000_0020, false,
            PreviewFeatures.isEnabled() ? Location.EMPTY_SET : Location.SET_CLASS,
            new Function<ClassFileFormatVersion, Set<Location>>() {
            @Override
            public Set<Location> apply(ClassFileFormatVersion cffv) {
-                return (cffv.compareTo(ClassFileFormatVersion.RELEASE_22) >= 0) &&
+               return (cffv.compareTo(ClassFileFormatVersion.RELEASE_23) >= 0) &&
                        PreviewFeatures.isEnabled() ? Location.EMPTY_SET : Location.SET_CLASS;
            }
        }),
```

**Pull request closed without merging**